File: codes/202603/15/LogEntry.cs
// LogEntry.cs
// 이 파일은 단일 로그 항목의 구조를 정의합니다.

using System;

namespace LogAnomalyDetection
{
    // 로그 엔트리를 나타내는 클래스입니다.
    public class LogEntry
    {
        // 로그가 발생한 시간입니다.
        public DateTime Timestamp { get; }
        // 로그의 레벨 (예: INFO, WARN, ERROR)입니다.
        public LogLevel Level { get; }
        // 로그 메시지의 출처 (예: 서비스 이름, 모듈 이름)입니다.
        public string Source { get; }
        // 실제 로그 메시지 내용입니다.
        public string Message { get; }

        // LogEntry 클래스의 새 인스턴스를 초기화합니다.
        public LogEntry(DateTime timestamp, LogLevel level, string source, string message)
        {
            Timestamp = timestamp;
            Level = level;
            Source = source;
            Message = message;
        }

        // LogEntry 객체를 문자열로 표현합니다.
        public override string ToString()
        {
            return $"[{Timestamp:yyyy-MM-dd HH:mm:ss}] [{Level}] [{Source}] {Message}";
        }
    }

    // 로그 레벨을 정의하는 열거형입니다.
    public enum LogLevel
    {
        INFO,   // 정보성 메시지
        WARN,   // 경고 메시지
        ERROR,  // 에러 메시지
        DEBUG   // 디버그 메시지
    }
}

File: codes/202606/23/LogEntry.java
// LogEntry.java
// 로그 파일의 각 항목을 나타내는 데이터 클래스입니다.

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class LogEntry {
    private LocalDateTime timestamp;
    private String level; // 예: INFO, WARN, ERROR
    private String source; // 예: com.example.MyService
    private String message;

    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    public LogEntry(LocalDateTime timestamp, String level, String source, String message) {
        this.timestamp = timestamp;
        this.level = level;
        this.source = source;
        this.message = message;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public String getLevel() {
        return level;
    }

    public String getSource() {
        return source;
    }

    public String getMessage() {
        return message;
    }

    @Override
    public String toString() {
        return "LogEntry{" +
               "timestamp=" + timestamp.format(FORMATTER) +
               ", level='" + level + ''' +
               ", source='" + source + ''' +
               ", message='" + message + ''' +
               '}';
    }
}
